core/modules/arp_responder.cc

```diff
@@ -0,0 +1,75 @@
+#include "arp_responder.h"
+
+#include "../utils/arp.h"
+
+using bess::utils::Arp;
+using bess::utils::be16_t;
+
+const Commands ArpResponder::cmds = {
+    {"add", "ArpResponderArg", MODULE_CMD_FUNC(&ArpResponder::CommandAdd), 0}};
+
+CommandResponse ArpResponder::CommandAdd(const bess::pb::ArpResponderArg &arg) {
+  be32_t ip_addr;
+  arp_entry entry;
+
+  if (!arg.ip().length()) {
+    return CommandFailure(EINVAL, "IP address is missing");
+  }
+  if (!bess::utils::ParseIpv4Address(arg.ip(), &ip_addr)) {
+    return CommandFailure(EINVAL, "Invalid IP Address: %s", arg.ip().c_str());
+  }
+
+  entry.mac_addr.FromString(arg.mac_addr());
+  entry.ip_addr = ip_addr;
+  entries[ip_addr] = entry;
```

Comment:
the mac is now filled as part of a check (see fix above) and its a bit hard to do it now